crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1762,6 +1797,84 @@ impl<T: R2D2Connection> IndexerStore for PgIndexerStore<T> {
     fn as_any(&self) -> &dyn StdAny {
         self
     }
+
+    /// Persist protocol configs and feature flags until the protocol version for the latest epoch
+    /// we have stored in the db, inclusive.
+    async fn persist_protocol_configs_and_feature_flags(&self) -> Result<(), IndexerError> {
+        // Fetch the chain id from db. We need a while loop here because it's possible that
+        // the first checkpoint has not beed indexed yet, so we need to wait for it.
+        let mut chain_id_opt;
+        while {
+            chain_id_opt = self.get_chain_identifier()?;
+            chain_id_opt.is_none()
+        } {}
+        let chain_id = ChainIdentifier::from(
+            CheckpointDigest::try_from(chain_id_opt.unwrap()).expect("Unable to convert chain id"),
+        );
+
+        // We only store protocol configs until the version for the latest epoch, because config
+        // values for future epochs, even when they already exist, are subject to change.
+        let end_version = self.get_latest_epoch_protocol_version()?.unwrap_or(1);
+        let mut all_configs = vec![];
+        let mut all_flags = vec![];
+
+        // Protocol version starts from 1.
+        let start_version = self.get_latest_protocol_config_version()?.unwrap_or(0) + 1;
+        info!(
+            "Persisting protocol configs with start_version: {}, end_version: {}",
+            start_version, end_version
+        );
+
+        // Gather all protocol configs and feature flags for all versions between start and end.
+        for version in start_version..=end_version {
```

Comment:
is this `for` for backfill, in other words, if indexer starts along with genesis, here each time we will always run only once?

Reply:
> if indexer starts along with genesis, here each time we will always run only once?

Yes. If indexer is in sync, this line will be run once per epoch so each time the loop will iterate at most once.

Reply:
ok i see, i suppose if we don't do this at indexer startup, then the first time we hit this loop we may be doing this for a while